README.md

```diff
@@ -273,9 +282,150 @@ dependencies:
+      "haskell"
+    ],
+    "initializationOptions": {
+      "languageServerHaskell": {
+      }
+    }
+  }
+}
+```
+
+#### LanguageClient-neovim
+
+##### vim-plug
+
+If you use [vim-plug](https://github.com/junegunn/vim-plug), then you can do this by e.g.,
+including the following line in the Plug section of your `init.vim` or `~/.vimrc`:
+
+```text
+Plug 'autozimu/LanguageClient-neovim', {
+    \ 'branch': 'next',
+    \ 'do': 'bash install.sh'
+    \ }
+```
+
+and issuing a `:PlugInstall` command within Neovim or Vim.
+
+##### Clone the LanguageClient-neovim repo
+
+As an alternative to using [vim-plug](https://github.com/junegunn/vim-plug) shown above, clone [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim)
+into `~/.vim/pack/XXX/start/`, where `XXX` is just a name for your "plugin suite".
+
+##### Sample `~/.vimrc`
+
+```vim
+set rtp+=~/.vim/pack/XXX/start/LanguageClient-neovim
+let g:LanguageClient_serverCommands = { 'haskell': ['haskell-language-server-wrapper', '--lsp'] }
+```
+
+You'll probably want to add some mappings for common commands:
+
+```vim
+nnoremap <F5> :call LanguageClient_contextMenu()<CR>
+map <Leader>lk :call LanguageClient#textDocument_hover()<CR>
+map <Leader>lg :call LanguageClient#textDocument_definition()<CR>
+map <Leader>lr :call LanguageClient#textDocument_rename()<CR>
+map <Leader>lf :call LanguageClient#textDocument_formatting()<CR>
+map <Leader>lb :call LanguageClient#textDocument_references()<CR>
+map <Leader>la :call LanguageClient#textDocument_codeAction()<CR>
+map <Leader>ls :call LanguageClient#textDocument_documentSymbol()<CR>
+```
+
+Use <kbd>Ctrl+x</kbd><kbd>Ctrl+o</kbd> (`<C-x><C-o>`) to open up the auto-complete menu,
+or for asynchronous auto-completion, follow the setup instructions on
+[LanguageClient](https://github.com/autozimu/LanguageClient-neovim).
+
+If you'd like diagnostics to be highlighted, add a highlight group for `ALEError`/`ALEWarning`/`ALEInfo`,
+or customize `g:LanguageClient_diagnosticsDisplay`:
+
+```vim
+hi link ALEError Error
+hi Warning term=underline cterm=underline ctermfg=Yellow gui=undercurl guisp=Gold
+hi link ALEWarning Warning
+hi link ALEInfo SpellCap
+```
+
+If you're finding that the server isn't starting at the correct project root,
+it may also be helpful to also specify root markers:
+
+```vim
+let g:LanguageClient_rootMarkers = ['*.cabal', 'stack.yaml']
+```
+
+### Using Haskell Language Server with Atom
+
+Make sure haskell-language-server and haskell-language-server-wrapper are installed, then install the two Atom packages [atom-ide-ui](https://atom.io/packages/atom-ide-ui) and [ide-haskell-hie](https://atom.io/packages/ide-haskell-hie),
+
+```bash
+$ apm install language-haskell atom-ide-ui ide-haskell-hie
+```
+
+The plugin ide-haskell-ide is designed to work with haskell-ide-engine by default, so you will have to put the path to haskell-language-server-wrapper in the configuration option `Absolute path to hie executable`.
```

Comment:
It would be great that an atom user could confirm that this tweak works

Reply:
I just tested and it works indeed.

Maybe we can rewrite/cleanup this part?
Proposal:

### Using Haskell Language Server with Atom

Install the following Atom packages:
- [Haskell language support](https://atom.io/packages/language-haskell)
- [Atom IDE UI](https://atom.io/packages/atom-ide-ui)
- [Haskell IDE Engine](https://atom.io/packages/ide-haskell-hie)

```shell
$ apm install language-haskell atom-ide-ui ide-haskell-hie
```

The Haskell IDE Engine package is designed to work with [Haskell IDE Engine](https://github.com/haskell/haskell-ide-engine) by default, so you need to change the executable path. Go to Settings / Packages / ide-haskell-hie / Settings / "Absolute path to hie executable" and replace the default value with `haskell-language-server`

If you want to use the wrapper to select the right GHC version for each project, also configure:
- Use the hie-wrapper: checked
- Use a custom hie wrapper: checked
- The path to your custom hie wrapper: `haskell-language-server`